### PR TITLE
CLI app project structure

### PR DESCRIPTION
**Pull request by iphilpot**
## All Submissions:
-------------------------------------
* [YES] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [NA] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [NA] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------
Introducing new behavior

## What is the new behavior?
-------------------------------------
Currently this only contains the start of the project structure for the CLI, a test that is only a pattern to follow once it's no just plumbing code, Makefile, and Readme. From a terminal, cd into the `cli` folder and run:

```make deps```
```make build```
```make run```
```make test```

We are currently only using go-cobra and go-testing packages just to get something committed. 

To use the cli:

`make build`

Look at help:

`./cobalt -h`

`./cobalt <cmd> -h`

Look at cmds:

`./cobalt setup --template "cobalt/azure-simple"`

*cmds should only print command issued to screen*

## Does this introduce a breaking change?
-------------------------------------
- [No]

## Any relevant logs, error output, etc?
-------------------------------------
- [No]

## Other information
-------------------------------------
- [NA]

### CODE REVIEW

File: cli/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Cobalt CLI
+
+Currently this only contains the start of the project structure, a test concept, Makefile and Readme. From a terminal, cd into the `cli` folder and run:
+
+```make deps```
```
Comment (nmiodice) [inline cli/README.md]: Is there any reason in particular that we should avoid having a single target which executes these in-order? As a new user I'd imagine that `make build` would do this all on my behalf.

File: cli/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+```cobalt setup --template "bedrock/azure-simple"```
+
+> The flag `--template` is specified using the following structure:
+>
+> `[github repo name]/[cobalt template name]`
+
+Test the current template in current directory:
+
+```cobalt run --docker```
```
Comment (nmiodice) [inline cli/README.md]: Can you elaborate on what will be run when this command is implemented? Does this actually build out resources?

Looking down further I see that it is a test runner. Should this be called something along the lines of `test`?

File: cli/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+# Cobalt Usage
+1. Create project dir: my_cobalt_proj
+2. `cd` to my_cobalt_proj
+3. Run `cobalt setup --template "cobalt/azure-simple"`
+4. Follow prompts for cloud provider info
+
+```
+ianphil@afropro.local [~/src/tmp/my_cobalt_proj]
+> tree -a
```
Comment (nmiodice) [inline cli/README.md]: TIL about tree!

File: cli/cmd/root.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+var (
+	cfgFile string
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "cobalt",
+	Short: "Cobalt is an opinionated tool for configuring cloud providers with terraform templates.",
+	Long: `
+
+	_________     ______        __________
```
Comment (nmiodice) [inline cli/cmd/root.go]: I dig it

File: cli/cmd/root.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+		// Find home directory.
+		home, err := homedir.Dir()
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// Search config in home directory with name ".cobalt" (without extension).
+		viper.AddConfigPath(home)
+		viper.SetConfigName(".cobalt")
```
Comment (erikschlegel) [inline cli/cmd/root.go]: NIT: you may want to consider moving `.cobalt` as a `DEFAULT_CONFIG_FILE` constant.

File: cli/README.md (hunk 2 above)
Comment (iphilpot) [reply cli/README.md]: Should be an easy refactor and I don't disagree. @erikschlegel for context...

It does nothing no so the docs, ATM, are shallow. I've not worked out the spec for the story yet so not really sure :-P

File: cli/README.md (hunk 1 above)
Comment (iphilpot) [reply cli/README.md]: I can create an "all" flag. Standard convention to break them out... good catch.

File: cli/Makefile [hunk 6]
```diff
@@ -0,0 +1,15 @@
+.PHONY: all test clean
```
[the remark this answers is not in the document]
Comment (iphilpot) [reply cli/Makefile]: This tells make to not look at the file system for file/dir when executing `make test`. It prevents name collisions (make will look for a file/folder by that name first before a flag). I don't really need all or clean in this.

File: cli/Gopkg.toml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+# Gopkg.toml example
```
[the remark this answers is not in the document]
Comment (iphilpot) [reply cli/Gopkg.toml]: I've added it and ran git rm on the files... should be good

**Comment by nmiodice** [comment]
FYI I believe I fixed the build issue you are running into, which is only rearing its ugly head now that we are running fork PRs.

For context, here is what I changed: https://github.com/microsoft/cobalt/pull/105/files#diff-5741e0c550b7539f2b44e35ba19fc8be

**Pull request merged**